tests/federation_room_join_partial_state_test.go

```diff
@@ -563,6 +565,128 @@ func TestPartialStateJoin(t *testing.T) {
 			t.Errorf("Did not find derek's m.room.member event in gappy /sync response: %s", err)
 		}
 	})
+
+	// regression test for https://github.com/matrix-org/synapse/issues/13001
+	//
+	// There was an edge case where, if we initially receive lots of events as outliers,
+	// and they then get de-outliered as partial state events,
+	t.Run("Resync works with many prev_events with partial state", func(t *testing.T) {
```

Comment:
Are we missing the other half of the comment?

Reply:
Also can we make the name of the test more specific? something along the lines of persisting chains of prev_events out of order maybe

Reply:
I've had a go at fixing the name of the test - perhaps it's better?

Reply:
Looks alright!